### PR TITLE
Use python3 rather than specific minor version (python3.9)

### PR DESCRIPTION
**Pull request by p-datadog**
Replaces #2315.

## Motivation

On current datadog ubuntu machines the python version is 3.10, not 3.9. Thus python3.9 binary is missing.

## Changes

This PR changes python3.9 reference to just python3.

If a minimum version of 3.9 is required I can add a check for this.

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] [Relevant label](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md) (`run-parametric-scenario`, `run-profiling-scenario`...) are presents
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
    * [ ] To R&P team: locally build and push the image to hub.docker.com 
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
    * [ ] Once merged, add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

**Comment by cbeauchesne** [comment]
converting to draft, waiting for the CI to be green for a review

File: utils/build/build.sh [hunk 1]
```diff
@@ -136,8 +136,20 @@ build() {
...
+                  local python_bin
+                  if python3 -V |egrep -q '^Python 3\.(9|10)\.'; then
+                      python_bin=python3
+                  else
+                      python_bin=python3.9
+                  fi
+                  
                   echo "Build virtual env"
-                  python3.9 -m venv venv
+                  python3 -m venv venv
```
Comment (smola) [inline utils/build/build.sh:152]:
```suggestion
                  "${python_bin}" -m venv venv
```

**Review by smola — CHANGES_REQUESTED** [review]

File: utils/build/build.sh (hunk 1 above)
Comment (cbeauchesne) [reply utils/build/build.sh:152]: is `python_bin` available on **all** our laptops (linux, windows, mac ...)

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
As it'll change the supporting python version from 3.9 to all 3.x matrx, can you add this matrix in `.github/workflows/ci.yml`, on `test_the_test` job ?

**Comment by cbeauchesne** [comment]
side question, do you have python 3.11 on your ubuntu ? 

I'm not very confortable to support several versions of python, big open door to troubles. If we can go directly to the target version (and the higher possible), I think we'll save some trouble later.

**Comment by cbeauchesne** [comment]
Closing this PR, as we are now using python 3.11

**Pull request closed without merging**